Copter: Add option to resume precland after reposiiton

rishabsingh3003 opened the pull request:
Currently, if the user were to reposition the vehicle during a Prec Land manually, it would cancel Prec Land and not resume it even if the target was in view (i.e Copter just does a normal land). 
By using PLND_OPTIONS parameter, one can now allow resuming prec land even after a manual reposition. 

This has been tested on SITL

khancyr commented on ArduCopter/mode.cpp:
```diff
@@ -668,6 +668,10 @@ void Mode::land_run_horizontal_control()
                 }
                 copter.ap.land_repo_active = true;
+            } else {
+                if (copter.ap.land_repo_active && copter.precland.allow_precland_after_reposition()) {
```
Copter.ap.land_repo_active is totally redundant here.

khancyr replied:
Probably need a guard to check that precland is available otherwise the copter.precland object won't exist

khancyr reviewed (changes requested):
Need a #if guard again precland to work.

Otherwise, the feature can be interesting and the implementation won't impact much the code if the option isn't enabled. So that is fine.

rishabsingh3003 commented:
@khancyr thanks! I have fixed those things, however, the RTL land_repo_active isn't related to precland in any way. For prec land, RTL calls the same controller, which we have already fixed in this PR.

khancyr commented:
yep, my bad on this, was looking at the wrong codebase. that is fine then as indeed descent_run() don't use precland yet

rmackay9 commented:
Merged thanks!

rmackay9 commented:
This is included in Copter-4.3.4-rc1
